### ISSUE TITLE
vim mode: Don't switch_mode when editor selection changed(not change by vim operator)

### ISSUE THREAD

**Issue by CharlesChen0823** [zed-industries/zed#14622]
### Description

currently, in `vim_mode`, vim will switch_mode when editor selections changed(not change by vim operator). this will cause some wried behaviour, e.g. when using `g l` for search all, `rename` using menu.

**Comment by Congyuwang** [comment]
I think like `g l` switching from NORMAL to VISUAL is necessary, and also when making selections, since you do not have selections in NORMAL mode. But I somehow do not like the switching to VISUAL when just mouse right clicking:

https://github.com/user-attachments/assets/ac3a03d1-bac6-4a13-9670-ecedeab9f507

**Comment by CharlesChen0823** [comment]
I cannot thinking which use case for `g l` switch `normal` mode to `visual` model? can you give me an example?

**Comment by Congyuwang** [comment]
Yeah you are probably right. You can also have multiple cursor in NORMAL model (which probably also work for `g l`), but just not range selection in NORMAL mode (like selecting a symbol), am I right?

**Comment by ConradIrwin** [comment]
I have been thinking about making `gl` keep you in normal mode and just jump to the same position in the current word. That might be better (and more like how add cursor below work).

If you want to take a pass at this feel free (or pair with me on it  https://calendly.com/conradirwin/pairing).

**Comment by notpeter** [comment]
@ConradIrwin / @CharlesChen0823 Is this still an issue? 

If so, can you edit the description to more explicitly expand upon the problem this issue is describing?

**Comment by ConradIrwin** [comment]
Closing for now. Also related is [#vimselec](https://github.com/zed-industries/zed/issues/34525).

I think it is (in general) an issue, and specifically we need to fix it for some actions; so lets track those more concretely.
